estuary/utils/story.py

```diff
@@ -195,9 +203,155 @@ def get_wait_times(self, results):
+                    artifact.__label__, id_num
+                )
+                continue
+
+            if artifact.__label__.endswith('KojiBuild'):
+                build = artifact
+
+            if artifact.__label__.endswith('Advisory'):
+                if artifact.state in ['SHIPPED_LIVE', 'DROPPED_NO_SHIP']:
+                    completion_time = getattr(artifact, timed_processes[artifact.__label__][1])
+                else:
+                    completion_time = datetime.utcnow()
+                if build:
+                    creation_time = artifact.attached_build_time(artifact, build)
+
+            # We do not want the processing time of the entire FreshmakerEvent, just the
+            # processing time until the displayed ContainerKojiBuild is created
+            elif artifact.__label__ == 'FreshmakerEvent':
+                if index != len(results) - 1:
+                    next_artifact = results[index + 1]
+                    completion_time = getattr(next_artifact,
+                                              timed_processes[next_artifact.__label__][0])
+                elif artifact.state_name in ['COMPLETE', 'SKIPPED', 'FAILED', 'CANCELED']:
+                    completion_time = getattr(artifact, timed_processes['FreshmakerEvent'][1])
+                    if completion_time is None:
+                        log.warning(
+                            'While calculating the total processing time, a %s with ID %s was '
+                            'encountered without a completion time or subsequent build.',
+                            artifact.__label__, id_num
+                        )
+                        continue
+                else:
+                    completion_time = datetime.utcnow()
+
+            else:
+                completion_time = getattr(artifact, timed_processes[artifact.__label__][1])
+                if not completion_time:
+                    completion_time = datetime.utcnow()
+
+            # Remove timezone info so that both are offset naive and thus able to be subtracted
+            creation_time = creation_time.replace(tzinfo=None)
+            completion_time = completion_time.replace(tzinfo=None)
+            processing_time = completion_time - creation_time
+
+            if processing_time.total_seconds() < 0:
+                id_num = getattr(artifact, artifact.unique_id_property + '_')
+                log.warning(
+                    'A negative processing time was calculated, with a %s with ID %s.',
+                    artifact.__label__, id_num
+                )
+
+            total += processing_time.total_seconds()
+
+        if total < 0:
+            id_num = getattr(artifact, artifact.unique_id_property + '_')
+            log.warning(
+                'A negative total processing time was calculated, in a story ending with a %s with '
+                'ID %s.',
+                artifact.__label__, id_num
+            )
+        return total
```

Comment:
If `if total < 0:` is `True`, should we just return 0? I don't know if the frontend would support negative seconds. This point is moot though depending on your decision on the comment:
https://github.com/release-engineering/estuary-api/pull/254/files#r306845653

Reply:
I realized that if we look at each individual processing time, and  check if each one is negative, we don't even need to check the total time because it cannot be negative, since all the negatives are skipped with each iteration.